Do not fail when cleanup after squashing is not possible

```
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/docker/api/client.py", line 229, in _raise_for_status
    response.raise_for_status()
  File "/usr/lib/python3.7/site-packages/requests/models.py", line 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 409 Client Error: Conflict for url: http+docker://localhost/v1.22/images/sha256:f210100c96aaa86d6ee7b74ca7b760e6dae1804b79839aa390cdb501a0538969?force=False&noprune=False
 
During handling of the above exception, another exception occurred:
 
Traceback (most recent call last):
  File "/usr/local/bin/cekit", line 11, in <module>
    load_entry_point('cekit', 'console_scripts', 'cekit')()
  File "/usr/local/lib64/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
...
docker.errors.APIError: 409 Client Error: Conflict ("b'conflict: unable to delete f210100c96aa (must be forced) - image is referenced in multiple repositories'")
make: *** [Makefile:9: build] Error 1
```

This issue can be reproduced with this project: https://github.com/spolti/quarkus-images/tree/quarkus-cekit-poc
But, remove the manual docker-squash command and the no-squash flag on the Makefile.

Full stacktrace:

```
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/docker/api/client.py", line 229, in _raise_for_status
    response.raise_for_status()
  File "/usr/lib/python3.7/site-packages/requests/models.py", line 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 409 Client Error: Conflict for url: http+docker://localhost/v1.22/images/sha256:f210100c96aaa86d6ee7b74ca7b760e6dae1804b79839aa390cdb501a0538969?force=False&noprune=False

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/cekit", line 11, in <module>
    load_entry_point('cekit', 'console_scripts', 'cekit')()
  File "/usr/local/lib64/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib64/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib64/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib64/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib64/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib64/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib64/python3.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/data/dev/sources/cekit/cekit/cli.py", line 106, in build_docker
    run_build(ctx, 'docker')
  File "/data/dev/sources/cekit/cekit/cli.py", line 282, in run_build
    run_command(ctx, builder_impl)
  File "/data/dev/sources/cekit/cekit/cli.py", line 251, in run_command
    Cekit(common_params).run(clazz, params)
  File "/data/dev/sources/cekit/cekit/cli.py", line 341, in run
    command.execute()
  File "/data/dev/sources/cekit/cekit/builder.py", line 60, in execute
    self.run()
  File "/data/dev/sources/cekit/cekit/builders/docker_builder.py", line 186, in run
    image_id = self._squash(docker_client, image_id)
  File "/data/dev/sources/cekit/cekit/builders/docker_builder.py", line 160, in _squash
    return squash.run()
  File "/usr/lib/python3.7/site-packages/docker_squash/squash.py", line 59, in run
    return self.squash(image)
  File "/usr/lib/python3.7/site-packages/docker_squash/squash.py", line 102, in squash
    self._cleanup()
  File "/usr/lib/python3.7/site-packages/docker_squash/squash.py", line 77, in _cleanup
    self.docker.remove_image(image_id, force=False, noprune=False)
  File "/usr/lib/python3.7/site-packages/docker/utils/decorators.py", line 19, in wrapped
    return f(self, resource_id, *args, **kwargs)
  File "/usr/lib/python3.7/site-packages/docker/api/image.py", line 480, in remove_image
    return self._result(res, True)
  File "/usr/lib/python3.7/site-packages/docker/api/client.py", line 235, in _result
    self._raise_for_status(response)
  File "/usr/lib/python3.7/site-packages/docker/api/client.py", line 231, in _raise_for_status
    raise create_api_error_from_http_exception(e)
  File "/usr/lib/python3.7/site-packages/docker/errors.py", line 31, in create_api_error_from_http_exception
    raise cls(e, response=response, explanation=explanation)
docker.errors.APIError: 409 Client Error: Conflict ("b'conflict: unable to delete f210100c96aa (must be forced) - image is referenced in multiple repositories'")
make: *** [Makefile:9: build] Error 1
```

--- a/docker_squash/squash.py
+++ b/docker_squash/squash.py
@@ -2,6 +2,8 @@
 
 import os
 
+import docker
+
 from distutils.version import StrictVersion
 
 from docker_squash.v1_image import V1Image
@@ -69,17 +71,24 @@ class Squash(object):
     def _cleanup(self):
         try:
             image_id = self.docker.inspect_image(self.image)['Id']
-        except:
-            self.log.warn("Could not get the image ID for %s image, skipping cleanup after squashing" % self.image)
+        except docker.errors.APIError as ex:
+            self.log.warn(
+                "Could not get the image ID for {} image: {}, skipping cleanup after squashing".format(
+                    self.image, str(ex)))
             return
 
-        self.log.info("Removing old %s image..." % self.image)
-        self.docker.remove_image(image_id, force=False, noprune=False)
-        self.log.info("Image removed!")
+        self.log.info("Removing old {} image...".format(self.image))
+
+        try:
+            self.docker.remove_image(image_id, force=False, noprune=False)
+            self.log.info("Image {} removed!".format(self.image))
+        except docker.errors.APIError as ex:
+            self.log.warn(
+                "Could not remove image {}: {}, skipping cleanup after squashing".format(self.image, str(ex)))
 
     def squash(self, image):
         # Do the actual squashing
-        new_image_id = image.squash()
+        new_image_id=image.squash()
 
         self.log.info("New squashed image ID is %s" % new_image_id)
 

--- a/tests/test_unit_squash.py
+++ b/tests/test_unit_squash.py
@@ -3,6 +3,8 @@ import mock
 import six
 import logging
 
+import docker
+
 from docker_squash.squash import Squash
 from docker_squash.image import Image
 from docker_squash.errors import SquashError
@@ -37,7 +39,30 @@ class TestSquash(unittest.TestCase):
     @mock.patch('docker_squash.squash.V2Image')
     def test_should_cleanup_after_squashing(self, v2_image):
         self.docker_client.inspect_image.return_value = {'Id': "abcdefgh"}
+
         squash = Squash(self.log, 'image', self.docker_client, load_image=True, cleanup=True)
         squash.run()
 
         self.docker_client.remove_image.assert_called_with('abcdefgh', force=False, noprune=False)
+        self.log.info.assert_any_call("Image image removed!")
+
+    @mock.patch('docker_squash.squash.V2Image')
+    def test_should_handle_cleanup_error_while_getting_image_id(self, v2_image):
+        self.docker_client.inspect_image.side_effect = docker.errors.APIError("Message")
+        
+        squash = Squash(self.log, 'image', self.docker_client, load_image=True, cleanup=True)
+        squash.run()
+
+        self.docker_client.remove_image.assert_not_called()
+        self.log.warn.assert_any_call("Could not get the image ID for image image: Message, skipping cleanup after squashing")
+
+    @mock.patch('docker_squash.squash.V2Image')
+    def test_should_handle_cleanup_error_when_removing_image(self, v2_image):
+        self.docker_client.inspect_image.return_value = {'Id': "abcdefgh"}
+        self.docker_client.remove_image.side_effect = docker.errors.APIError("Message")
+
+        squash = Squash(self.log, 'image', self.docker_client, load_image=True, cleanup=True)
+        squash.run()
+
+        self.log.info.assert_any_call("Removing old image image...")
+        self.log.warn.assert_any_call("Could not remove image image: Message, skipping cleanup after squashing")

